### ISSUE TITLE
Replace `setup_requires` by `pyproject.toml`

### ISSUE THREAD

**Issue by pastewka** [ContactEngineering/SurfaceTopography#167]
See: https://www.python.org/dev/peps/pep-0517/#build-requirements

**Comment by mcrot** [comment]
See also https://github.com/pypa/pip/issues/8368.
When building `SurfaceTopography` in a Docker container for TopoBank, I always get this:
```
   Running setup.py install for surfacetopography: started                                                                                                                             
    Running setup.py install for surfacetopography: finished with status 'done'                                                                                                         
  DEPRECATION: surfacetopography was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. A possible replacement is to fix the wheel build issue reported above. You can find discussion regarding this at https://github.com/pypa/pip/issues/8368. 
```
Maybe the compilation will fail in near future because of that, so this should get some priority I think.
Shall we also work with milestones in this project?

**Comment by mcrot** [comment]
I also have no experience with this, but I'm interested to know how it works. I can this pick up in July.

**Comment by mcrot** [comment]
Has this already been done via #246 or this there still something needed here?

**Comment by pastewka** [comment]
Essentially addressed by #246
